core/src/main/kotlin/org/eln2/sim/thermal/main.kt

```diff
@@ -0,0 +1,181 @@
+import kotlin.math.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.*
+
+
+/**
+ * represents a thermal interface between two objects. At each tick, the flow can be marked as computed.
+ */
+class ThermalInterface {
+
+    var mutex = Mutex()
+    var R = 1.0 // heat resistance at the interface
+    var heatTranfer = 0.0 // heat transfer during a simulation tick through this interface
+
+    var lastTickComputed = 0 // number of the last tick where the heat transfer was computed. Tick number switch between 0 and 1. This is here to prevent computing twice the heat trasnfered when not needed
```

Comment:
I'll think of something for the lastTickComputed variable since IProcess only have a time delta.

Should I make the rest private or whatever Kotlin equivalent is?

Reply:
`private` will work until you try to access it with tests, if you need that, use `internal`.